### PR TITLE
2.6.0 filter categories if helpdesk, filter categories with depth and type criterias

### PR DESCRIPTION
**Pull request by btry**

### CODE REVIEW

**Comment by btry** [comment]
see #689

**Comment by btry** [comment]
This PR should be reworked after #690 is merged (possible conflicts by the way)

**Comment by wawax** [comment]
After a first test : 

1/ there's an issue when displaying a category field. 

Fatal error: Uncaught Error: Class name must be a valid object or a string in /var/www/html/glpi/plugins/formcreator/inc/fields/dropdownfield.class.php:42 Stack trace: #0 /var/www/html/glpi/plugins/formcreator/inc/field.class.php(29): PluginFormcreatorDropdownField->displayField(true) #1 /var/www/html/glpi/plugins/formcreator/inc/fields.class.php(106): PluginFormcreatorField->show(true) #2 /var/www/html/glpi/plugins/formcreator/inc/form.class.php(871): PluginFormcreatorFields::showField(Array, NULL) #3 /var/www/html/glpi/plugins/formcreator/inc/form.class.php(480): PluginFormcreatorForm->displayUserForm(Object(PluginFormcreatorForm)) #4 /var/www/html/glpi/inc/commonglpi.class.php(483): PluginFormcreatorForm::displayTabContentForItem(Object(PluginFormcreatorForm), '1', '') #5 /var/www/html/glpi/ajax/common.tabs.php(97): CommonGLPI::displayStandardTab(Object(PluginFormcreatorForm), 'PluginFormcreat...', '', Array) #6 {main} thrown in /var/www/html/glpi/plugins/formcreator/inc/fields/dropdownfield.class.php on line 42

2/ when editing a category field, the default value is not displayed. It's only displayed during field creation.

**Comment by btry** [comment]
2nd issue should be fixed, but I can't investgate the first one as formcreator/inc/fields/dropdownfield.class.php:42 contains only a closing curling bracket for me.

I believe you aplied the diff of this PR on a released version. I need confirmation to continue investigation.

**Comment by wawax** [comment]
Yes I applied this PR on the latest release @btry

File: inc/fields/dropdownfield.class.php [hunk 1]
```diff
@@ -16,6 +21,25 @@ public function displayField($canEdit = true) {
...
+                  $dparams['condition'] = "`is_helpdeskvisible` = '1'";
+               }
+               switch ($decodedValues['show_ticket_categories']) {
+                  case 'request':
+                     $dparams['condition'] = "`is_request` = '1'";
+                     break;
+                  case 'incident':
+                     $dparams['condition'] = "`is_incident` = '1'";
+                     break;
+                  case 'both':
```
Comment (orthagh) [inline inc/fields/dropdownfield.class.php]: is this case needed ?

File: inc/fields/dropdownfield.class.php [hunk 2]
```diff
@@ -16,6 +21,25 @@ public function displayField($canEdit = true) {
 
             if ($itemtype == "User") {
                $dparams['right'] = 'all';
+            } else if ($itemtype == "ITILCategory") {
+               if (isset ($_SESSION['glpiactiveprofile']['interface'])
+                   && $_SESSION['glpiactiveprofile']['interface'] == 'helpdesk') {
+                  $dparams['condition'] = "`is_helpdeskvisible` = '1'";
```
Comment (orthagh) [inline inc/fields/dropdownfield.class.php]: this ``condition`` key could be erased by the filtering bellow

**Comment by btry** [comment]
in a target ticket category from a question is broken (probably due to introduction of JSON format)

**Pull request merged**